### ISSUE TITLE
SQL Telemetry throws error when using Azure SQL Server.

### ISSUE THREAD

**Issue by kilsbo** [microsoft/ApplicationInsights-Java#722]
### Expected behavior
SQL Telemetry sent from agent of JDBC dependency usage, using Spring Boot + AppInsight Agent + Azure SQL Server.

### Actual behavior
In the class CoreAgentNotificationsHandler, in method fetchExplainQuery() on (or around, if changed) line 547: `rs = explain.executeQuery("EXPLAIN " + commandName);` is always executed, and will not work on Azure SQL Server.

### To Reproduce
Configure an Agent when using Spring JPA Repositories, Hibernate and Azure SQL Server, other SQL Server editions might experience the same issue.

### System information
Please provide the following information:
 - SDK Version: OpenJDK 8
 - OS type and version: Docker image: openjdk:jre-alpine
 - Application Server type and version (if applicable): Tomcat
 - Using spring-boot? Yes
 - Additional relevant libraries (with version, if applicable): Hibernate/Spring JPA Repositories.

### Logs
Message: `Message    Could not find stored procedure 'EXPLAIN'`

```
com.microsoft.sqlserver.jdbc.SQLServerException:
   at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError (SQLServerException.java258)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult (SQLServerStatement.java1535)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.doExecuteStatement (SQLServerStatement.java845)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement$StmtExecCmd.doExecute (SQLServerStatement.java752)
   at com.microsoft.sqlserver.jdbc.TDSCommand.execute (IOBuffer.java7151)
   at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand (SQLServerConnection.java2478)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand (SQLServerStatement.java219)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement (SQLServerStatement.java199)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeQuery (SQLServerStatement.java654)
   at com.microsoft.applicationinsights.internal.agent.CoreAgentNotificationsHandler.fetchExplainQuery (CoreAgentNotificationsHandler.java543)
   at com.microsoft.applicationinsights.internal.agent.CoreAgentNotificationsHandler.sendSQLTelemetry (CoreAgentNotificationsHandler.java476)
   at com.microsoft.applicationinsights.internal.agent.CoreAgentNotificationsHandler.report (CoreAgentNotificationsHandler.java394)
   at com.microsoft.applicationinsights.internal.agent.CoreAgentNotificationsHandler.finalizeMethod (CoreAgentNotificationsHandler.java387)
   at com.microsoft.applicationinsights.internal.agent.CoreAgentNotificationsHandler.methodFinished (CoreAgentNotificationsHandler.java234)
   at com.microsoft.applicationinsights.agent.internal.coresync.impl.ImplementationsCoordinator.methodFinished (ImplementationsCoordinator.java266)
   at com.microsoft.sqlserver.jdbc.SQLServerStatement.execute (SQLServerStatement.java731)
   at org.apache.tomcat.jdbc.pool.PooledConnection.validate (PooledConnection.java532)
   at org.apache.tomcat.jdbc.pool.PooledConnection.validate (PooledConnection.java443)
   at org.apache.tomcat.jdbc.pool.ConnectionPool.borrowConnection (ConnectionPool.java802)
   at org.apache.tomcat.jdbc.pool.ConnectionPool.borrowConnection (ConnectionPool.java651)
   at org.apache.tomcat.jdbc.pool.ConnectionPool.getConnection (ConnectionPool.java198)
   at org.apache.tomcat.jdbc.pool.DataSourceProxy.getConnection (DataSourceProxy.java132)
   at org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.getConnection (DatasourceConnectionProviderImpl.java122)
   at org.hibernate.internal.AbstractSessionImpl$NonContextualJdbcConnectionAccess.obtainConnection (AbstractSessionImpl.java386)
   at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded (LogicalConnectionManagedImpl.java87)
   at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.getPhysicalConnection (LogicalConnectionManagedImpl.java112)
   at org.hibernate.internal.SessionImpl.connection (SessionImpl.java489)
   at sun.reflect.GeneratedMethodAccessor120.invoke
   at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java43)
   at java.lang.reflect.Method.invoke (Method.java498)
   at org.springframework.util.ReflectionUtils.invokeMethod (ReflectionUtils.java216)
   at org.springframework.util.ReflectionUtils.invokeMethod (ReflectionUtils.java201)
   at org.springframework.orm.jpa.vendor.HibernateJpaDialect$HibernateConnectionHandle.doGetConnection (HibernateJpaDialect.java414)
   at org.springframework.orm.jpa.vendor.HibernateJpaDialect.beginTransaction (HibernateJpaDialect.java177)
   at org.springframework.orm.jpa.JpaTransactionManager.doBegin (JpaTransactionManager.java380)
   at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction (AbstractPlatformTransactionManager.java373)
   at org.springframework.transaction.interceptor.TransactionAspectSupport.createTransactionIfNecessary (TransactionAspectSupport.java461)
   at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction (TransactionAspectSupport.java277)
   at org.springframework.transaction.interceptor.TransactionInterceptor.invoke (TransactionInterceptor.java96)
   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed (ReflectiveMethodInvocation.java179)
   at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke (PersistenceExceptionTranslationInterceptor.java136)
   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed (ReflectiveMethodInvocation.java179)
   at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke (CrudMethodMetadataPostProcessor.java133)
   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed (ReflectiveMethodInvocation.java179)
   at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke (ExposeInvocationInterceptor.java92)
   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed (ReflectiveMethodInvocation.java179)
   at org.springframework.data.repository.core.support.SurroundingTransactionDetectorMethodInterceptor.invoke (SurroundingTransactionDetectorMethodInterceptor.java57)
   at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed (ReflectiveMethodInvocation.java179)
   at org.springframework.aop.framework.JdkDynamicAopProxy.invoke (JdkDynamicAopProxy.java213)
   at com.sun.proxy.$Proxy134.findAll
   at com.company.service.StatusServiceImpl.findStatus (StatusServiceImpl.java59)
```

**Comment by dhaval24** [comment]
@kilsbo could you please tell which version of SQL Server you are using? Further I hope you are using 2.1.2 version AI-Agent. Please turn on SDK Logs and Agent logs by adding this property to the application.properties file:

`azure.application-insights.logger.level=trace`

and add <AgentLogger /> to AI-Agent.xml and upload the logs for further analysis.

**Comment by cy33hc** [comment]
@dhaval24 We have also enabled SQL telemetry with Azure SQL Server and getting the same exception. I'm on AI 2.1.1

**Comment by dhaval24** [comment]
@cy33hc would you please try to upgrade to the latest AI SDK (2.4.0-BETA) and see if your issue persists?

**Comment by cy33hc** [comment]
@dhaval24 I doubt it's fixed since the master code branch still executing the "EXPLAIN" cmd which doesn't exists in Azure SQL Server.

**Comment by kilsbo** [comment]
I appreciate the response to my initial post in this matter.

However, I've pointed out in the source code what the problem actually is, and the moment when our application hit that part of this library code, creating the Exception I've given. Therefore I'm beyond the phase where I'm analyzing logs.

Intuition tells me that it was assumed that SQL Server was not the obvious database choice when using this framework from an Java application. I get that. It's most likely MySQL, and maybe others.

We've since moved on to serverless architecture, since the tooling there for telemetry is on another level than hooking up agents to the JVM, so I cannot contribute more to this matter, I'm afraid.
